openid-4-verifiable-presentations-1_0.md

```diff
@@ -1900,88 +1900,137 @@ these constraints. The Wallet is not controlled by the Verifier and the Verifier
 MUST perform its own security checks on the returned Credentials and
 Presentations.
 
-## DCQL Value Matching {#dcql-value-matching}
+# Privacy Considerations {#privacy-considerations}
+
+Many privacy considerations are specific to the Credential format and associated proof type used in a particular Presentation.
+
+This section focuses on privacy considerations specific to the presentation protocol while also addressing cross-cutting concerns related to credential formats, Wallet behavior, and Verifier practices.
+
+Wallet providers and Verifiers need to take into account privacy considerations in this section to mitigate the risks of
+data leakage, user tracking, and other privacy harms. These considerations inform the development of privacy risk analysis.
+
+## User Consent {#user_consent_and_choice}
+
+Wallets SHOULD obtain explicit, informed consent from the End-User before releasing any Verifiable Credential or Presentation to a Verifier.
+
+If an error prevents the Wallet from honoring a request (e.g., missing Credentials or mismatched claim values), the Wallet SHOULD inform the End-User in a privacy-preserving way.
+
+Transaction history and data within the Wallet SHOULD NOT be accessible to anyone other than the End-User, unless the End-User has given consent or there is another legal basis to do so.
+
+## Privacy Notice {#privacy_notice}
+
+Wallets SHOULD make their privacy notices readily available to the End-User.
+
+## Purpose Legitimacy {#purpose_legitimacy_and_specification}
+
+The Verifier SHOULD ensure that information collection purpose is sufficiently specific and communicated before collection. For example, the purpose is shown to the End-User before the presentation request is sent to the Wallet.
+
+If the Wallet has indications that the Verifier is requesting data that it is not entitled to, the Wallet SHOULD warn the End-User and potentially stop processing.
```

Comment:
```suggestion
If the Wallet has indications that the Verifier is requesting data that it is not entitled to, the Wallet SHOULD warn the End-User.
```

I don't know how to interpret "potentially stop processing" here and I feel this might lead to interoperability issues if we include it. If we think the text without this is insufficient then perhaps we should consider saying something like sending an error in response.

Reply:
I think what I'm saying is that "stop processing" is ambiguous and could lead to interoperability issues. Even without this text it remains in the wallets discretion to do what it likes beyond warning the user.

Reply:
i don't think anything in privacy considerations section leads to the interoperability issues. entire section describes how to implement what is already interoperable in a privacy preserving manner.